web/src/beta/features/PluginPlayground/Code/index.tsx

```diff
@@ -9,37 +16,54 @@ type Props = {
   executeCode: () => void;
 };
 
-const getLanguageByFileExtension = (fileTitle: string) => {
-  const ext = fileTitle.split(".").pop();
-  switch (ext) {
-    case "js":
-      return "javascript";
-    case "yml":
-    case "yaml":
-      return "yaml";
-    default:
-      return "plaintext";
-  }
-};
-
 const Code: FC<Props> = ({
   fileTitle,
   sourceCode,
   onChangeSourceCode,
   executeCode
 }) => {
+  const ediableHtmlSourceCode = useMemo(
+    () => extractHtmlFromSourceCode(sourceCode),
+    [sourceCode]
+  );
+
+  const onSubmitHtmlEditor = useCallback(
+    (newHtml: string) => {
+      onChangeSourceCode(injectHtmlIntoSourceCode(newHtml, sourceCode));
+    },
+    [sourceCode, onChangeSourceCode]
+  );
+
+  const [isOpenedHtmlEditor, setIsOpenedHtmlEditor] = useState(false);
+
   return (
-    <Wrapper>
-      <Header>
-        <Button icon="playRight" iconButton onClick={executeCode} />
-        <p>Widget</p>
-      </Header>
-      <CodeInput
-        language={getLanguageByFileExtension(fileTitle)}
-        value={sourceCode}
-        onChange={onChangeSourceCode}
-      />
-    </Wrapper>
+    <>
+      <Wrapper>
+        <Header>
+          <Button icon="playRight" iconButton onClick={executeCode} />
+          <Button
+            icon="pencilSimple"
+            title="HTML Editor"
+            disabled={ediableHtmlSourceCode === undefined}
```

Comment:
Maybe we could have this as `disabled={!editableHtmlSourceCode}`. Let me know your thoughts.

Reply:
Thank you. I originally wrote it this way with the strict intent of only targeting undefined. However, after considering it further, I realized that enabling disabled for empty strings as well would be appropriate, so I’ve revised it as you suggested.

https://github.com/reearth/reearth-visualizer/pull/1236/commits/8c39c0d0b42595a06a579e9bccbad1f3c84e728c